### DISCUSSION TITLE
Limit characters per column

### DISCUSSION THREAD

**Discussion by OCRAMIS** [Power-Components/livewire-powergrid#241] [Q&A] [answered]
Hi,

Is there any way to limit the number of characters ( display text ) per column? I am looking for something similar to 
**->limit(min/max)** . For instance, if I have blog that would have large number of characters in title and description. Those columns need to be reduced to around 50-75 characters.

Thank you in advance.

**Comment by dansysanalyst** [comment, marked as answer]
Hello there,

You can use a closure to create an excerpt for your `text` column.

For example, imagine you have a database table `poems` with the following data:

|  ID | TITLE  |  AUTHOR   | TEXT
| ------------ | ------------ | ------------ | ------------ |
|  1 | Tabacaria  | Fernando Pessoa  |   I am nothing  I shall never be anything I cannot want to be anything. Aside from that, I have within me all the dreams of the world. Windows of my room, The room of one of the world's millions nobody knows about (And if they knew about me, what would they know?) You open onto the mystery of a street continually crossed by people, A street inaccessible to any thought, Real, impossibly real, certain, unknowingly certain, With the mystery of things beneath the stones and beings, With death making the walls damp and the hair of men white, With Destiny driving the wagon of everything down the road of nothing.    |

You can do something like:

```php
<?php

use Illuminate\Support\Str;

//...

 public function addColumns(): ?PowerGridEloquent
    {
        return PowerGrid::eloquent()
            ->addColumn('id')
            ->addColumn('title')
            ->addColumn('author')
            ->addColumn('text_preview', function (Poem $poem) {
                return  Str::words($poem->text, 10); //Gets first 10 words
            });
    }
```

The outcome will be:

|  ID | TITLE  |  AUTHOR   | TEXT_PREVIEW
| ------------ | ------------ | ------------ | ------------ |
|  1 | Tabacaria  | Fernando Pessoa  |   I am nothing  I shall never be anything I cannot...   |


Read more here:

https://livewire-powergrid.com/#/table/add-columns?id=closure-examples


---

⭐ Did it solve your problem? Consider Starring this Repository!

**Comment by OCRAMIS** [comment]
@dansysanalyst,

Thank you very much for your quick response.
Your suggestion worked for me.
I modified it as follows:
->addColumn('text_preview', function (Poem $poem) {
                return  Str::limit($poem->text, '10'); //Gets first 10 characters

Best Regards.
